### PR TITLE
refac(math): move binary fields

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR moves code related to binary fields to `tachyon/math/finite_fields/binary_fields`.

### CODE REVIEW

**Review by ashjeong — APPROVED** [review]
LGTM

**Review by Insun35 — APPROVED** [review]
LGTM

File: tachyon/math/finite_fields/binary_fields/binary_fields_unittest.cc [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 #include "tachyon/math/finite_fields/binary_fields/binary_fields.h"
 
+#include "gtest/gtest.h"
+
 namespace tachyon::math {
 
 namespace {
 template <typename BinaryField>
-class BinaryFieldTest : public testing::Test {
+class BinaryFieldsTest : public testing::Test {
```
Comment (TomTaehoonKim) [inline tachyon/math/finite_fields/binary_fields/binary_fields_unittest.cc:9]: Did you intend to rename? Also the file name right?

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by dongchangYoo — APPROVED** [review]
LGTM

File: tachyon/math/finite_fields/binary_fields/binary_fields_unittest.cc (hunk 1 above)
Comment (chokobole) [reply tachyon/math/finite_fields/binary_fields/binary_fields_unittest.cc:9]: Oh yes, this is because the file name is `binary_fields_unittest.cc`.

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Pull request merged**